lib/module.js

```diff
@@ -88,9 +88,18 @@ function readPackage(requestPath) {
 function tryPackage(requestPath, exts) {
   var pkg = readPackage(requestPath);
 
-  if (!pkg) return false;
+  if (!(pkg && pkg.main)) return false;
+
+  var resolved = pkg['_resolved'],
```

Comment:
You mean the _shasum field?  Same issue: what if it's not unique, e.g., because someone accidentally copy-pasted it from one package.json to another?

You could perhaps take the SHA-1 checksum of the whole package.json file but that requires that node is compiled with openssl support.  You can add workarounds for the non-openssl case but I don't know, that gets complex awfully fast.